### PR TITLE
fix(save-responses): [nan-1827] save batchSave to namespace model Map and save input for action

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Handle multi model syncs for tests and save the action input to a file

## Issue ticket number and link
NAN-1827

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/cli/lib/services/dryrun.service.ts [hunk 1]
```diff
@@ -237,6 +237,17 @@ export class DryRunService {
                     normalizedInput = actionInput;
                 }
             }
+
+            if (options.saveResponses) {
+                const responseDirectoryPrefix = process.env['NANGO_MOCKS_RESPONSE_DIRECTORY'] ?? '';
+                const directoryName = `${responseDirectoryPrefix}${providerConfigKey}`;
+                if (!fs.existsSync(`${directoryName}/mocks/${syncName}`)) {
+                    fs.mkdirSync(`${directoryName}/mocks/${syncName}`, { recursive: true });
+                }
```
Comment (bodinsamuel) [inline packages/cli/lib/services/dryrun.service.ts]: ensureDirectoryExists()

File: packages/cli/lib/services/dryrun.service.ts [hunk 2]
```diff
@@ -364,6 +375,15 @@ export class DryRunService {
                     resultOutput.push(chalk.gray('no output'));
                 } else {
                     console.log(JSON.stringify(results.response, null, 2));
+                    if (options.saveResponses) {
+                        const responseDirectoryPrefix = process.env['NANGO_MOCKS_RESPONSE_DIRECTORY'] ?? '';
+                        const directoryName = `${responseDirectoryPrefix}${providerConfigKey}`;
+                        if (!fs.existsSync(`${directoryName}/mocks/${syncName}`)) {
+                            fs.mkdirSync(`${directoryName}/mocks/${syncName}`, { recursive: true });
+                        }
```
Comment (bodinsamuel) [inline packages/cli/lib/services/dryrun.service.ts]: ensureDirectoryExists()

File: packages/cli/lib/services/dryrun.service.ts [hunk 3]
```diff
@@ -403,17 +423,23 @@ export class DryRunService {
...
-                    }
-
-                    if (nangoProps.rawDeleteOutput) {
-                        const filePath = `${directoryName}/mocks/${syncName}/batchDelete.json`;
-                        fs.writeFileSync(filePath, JSON.stringify(nangoProps.rawDeleteOutput, null, 2));
+                    if (scriptInfo?.output) {
+                        for (const model of scriptInfo.output) {
+                            if (!fs.existsSync(`${directoryName}/mocks/${syncName}/${model}`)) {
+                                fs.mkdirSync(`${directoryName}/mocks/${syncName}/${model}`, { recursive: true });
+                            }
```
Comment (bodinsamuel) [inline packages/cli/lib/services/dryrun.service.ts]: ensureDirectoryExists

**Review by bodinsamuel — APPROVED** [review]

File: packages/cli/lib/services/dryrun.service.ts (hunk 1 above)
Comment (khaliqgant) [reply packages/cli/lib/services/dryrun.service.ts]: https://github.com/NangoHQ/nango/pull/2795/commits/2d82b320f7073efe111bb1eeccb12b1a32f2785c

File: packages/cli/lib/services/dryrun.service.ts (hunk 2 above)
Comment (khaliqgant) [reply packages/cli/lib/services/dryrun.service.ts]: https://github.com/NangoHQ/nango/pull/2795/commits/2d82b320f7073efe111bb1eeccb12b1a32f2785c

File: packages/cli/lib/services/dryrun.service.ts (hunk 3 above)
Comment (khaliqgant) [reply packages/cli/lib/services/dryrun.service.ts]: https://github.com/NangoHQ/nango/pull/2795/commits/2d82b320f7073efe111bb1eeccb12b1a32f2785c

**Pull request merged**